plugins/destination/postgresql/client/transformer.go

```diff
@@ -146,7 +146,7 @@ func transformArr(arr arrow.Array) []any {
 			nested := array.NewSlice(a.ListValues(), start, end)
 			pgArr[i] = transformArr(nested)
 		case *types.JSONArray:
-			pgArr[i] = a.Storage().(*array.Binary).Value(i)
+			pgArr[i] = stripNullsFromMarshalledJson(a.Storage().(*array.Binary).Value(i))
```

Comment:
IMO we could use the `GetOneForMarshal` method here

Reply:
You're right (same thing though)